app/controllers/session_controller.rb

```diff
@@ -10,6 +10,8 @@ class SessionController < ApplicationController
 
   requires_login only: [:second_factor_auth_show, :second_factor_auth_perform]
 
+  allow_in_staff_writes_mode :create
```

Comment:
I think it would be good to add an exception for `UsersController#admin_login` as well.

Ideally we would also make sure that SSO continues to function as well. Good news is that DiscourseConnect uses GET requests, so that should be safe. But for Omniauth-based auth, I think we'll need to add an exception for `OmniauthCallbacksController#confirm_request` and `OmniauthCallbacksController#complete`